File: ruby/example_code/s3/hello/hello_s3.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# snippet-start:[ruby.hello_s3]

# frozen_string_literal: true

# S3Manager is a class responsible for managing S3 operations
# such as listing all S3 buckets in the current AWS account.
class S3Manager
  def initialize(client)
    @client = client
    @logger = Logger.new($stdout)
  end

  # Lists and prints all S3 buckets in the current AWS account.
  def list_buckets
    @logger.info('Here are the buckets in your account:')

    response = @client.list_buckets

    if response.buckets.empty?
      @logger.info("You don't have any S3 buckets yet.")
    else
      response.buckets.each do |bucket|
        @logger.info("- #{bucket.name}")
      end
    end
  rescue Aws::Errors::ServiceError => e
    @logger.error("Encountered an error while listing buckets: #{e.message}")
  end
end

if $PROGRAM_NAME == __FILE__
  s3_client = Aws::S3::Client.new
  manager = S3Manager.new(s3_client)
  manager.list_buckets
end

# snippet-end:[ruby.hello_s3]

File: python/example_code/s3/hello/hello_s3.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# snippet-start:[python.example_code.s3.Hello]

import boto3

def hello_s3():
    """
    Demonstrates how to list the Amazon S3 buckets in your AWS account using the AWS SDK for Python.
    """
    # Create an S3 client
    s3 = boto3.client('s3')

    # Use the S3 client to list the buckets
    response = s3.list_buckets()

    # Print the names of all the buckets
    print(f"Your Amazon S3 buckets are:")
    for bucket in response['Buckets']:
        print(f"  {bucket['Name']}")

if __name__ == '__main__':
    hello_s3()

# snippet-end:[python.example_code.s3.Hello]